modules/explorer-core/src/main/scala/org/ergoplatform/explorer/protocol/utils.scala

```diff
@@ -64,4 +74,33 @@ object utils {
     }.toEither
       .leftMap(e => ErgoTreeDeserializationFailed(bytes, Option(e.getMessage)))
       .toRaise
+
+  @inline def ergoTreeToBytes[
+    F[_]: ContravariantRaise[*[_], ErgoTreeSerializationFailed]: Applicative
+  ](ergoTree: ErgoTree): F[Array[Byte]] =
+    Try {
+      ergoTree.bytes
+    }.toEither
+      .leftMap(e => ErgoTreeSerializationFailed(ergoTree, Option(e.getMessage)))
+      .toRaise
+
+  /** Extracts ErgoTree's template (serialized tree with placeholders instead of values)
+    * @param ergoTree ErgoTree
+    * @return serialized ErgoTree's template
+    */
+  def ergoTreeTemplateBytes[F[_]: ContravariantRaise[
+    *[_],
+    ErgoTreeDeserializationFailed
+  ]: ContravariantRaise[*[_], ErgoTreeSerializationFailed]: Monad](
```

Comment:
Please extend `ErgoTreeDeserializationFailed` and `ErgoTreeSerializationFailed` from more general parent and use it in the annotation.